src/parser/parser.yy

```diff
@@ -400,46 +397,143 @@ base_expression
     | KW_NULL {
         $$ = new ConstantExpression(NullType::__NULL__);
     }
-    | input_ref_expression {
+    | name_label {
+        $$ = new LabelExpression($1);
+    }
+    | INTEGER {
+        $$ = new ConstantExpression($1);
+    }
+    | compound_expression {
         $$ = $1;
     }
-    | src_ref_expression {
+    | MINUS {
```

Comment:
Will this make `-` a valid expression in parser?

Reply:
> Will this make `-` a valid expression in parser?

Actually it is a mid-rule action after the keyword `MINUS`